pkg/messaging/v1/invoke_method_request.go

```diff
@@ -106,6 +106,19 @@ func (imr *InvokeMethodRequest) WithHTTPExtension(verb string, querystring strin
 	return imr
 }
 
+// WithCustomMetdata applies a metadata map to a InvokeMethodRequest
+func (imr *InvokeMethodRequest) WithCustomHTTPMetadata(md map[string]string) *InvokeMethodRequest {
+	for k, v := range md {
+		if imr.r.Metadata == nil {
+			imr.r.Metadata = make(map[string]*internalv1pb.ListStringValue)
+		}
+
+		imr.r.Metadata[k] = &internalv1pb.ListStringValue{Values: []string{v}}
```

Comment:
We are mapping from pubsub metadata of type `map[string][string` which doesn't support duplicate keys to invoke method request metadata `map[string]*internalv1pb.ListStringValue` which does supports duplicate keys. If we have any pubsub that require more complex metadata we might need to account for this.